examples/failure_and_success_hooks/run.py

```diff
@@ -0,0 +1,76 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import random
+
+from zenml.hooks import alerter_failure_hook, alerter_success_hook
+from zenml.pipelines import pipeline
+from zenml.steps import BaseParameters, Output, StepContext, step
+
+
+def pipeline_success_hook(context: StepContext) -> None:
+    print(f"Step {context.step_name} succeeded!")
+
+
+class HookParams(BaseParameters):
+    """Parameters to define which hook to trigger"""
+
+    fail: bool = True
+
+
+@step(on_failure=alerter_failure_hook, on_success=alerter_success_hook)
+def get_first_num(params: HookParams) -> int:
```

Comment:
All our examples follow the recommended structure of having a `steps` and `pipelines` subdirectory which define those classes, I think we should stay consistent and also do that for this example